packages/client/src/components/Location/LocationFinder.tsx

```diff
@@ -65,109 +65,110 @@ const LocationFinder: FunctionComponent<LocationFinderProps> = ({
   const [autoSuggestValue, setAutoSuggestValue] = useState<string>("");
-  const [touched, setTouched] = useState<{ [key: string]: boolean }>({});
+  const [focus, setFocus] = useState<boolean>(false);
+
+  const checkPostalCodeErrors = () => {
+    setPostalCodeError(
+      postalCode && !isValidPostalcode(postalCode)
+        ? t("common.no valid postalcode")
+        : ""
+    );
+  };
+  const checkHouseNumberErrors = () => {
+    setHouseNumberError(
+      houseNumber?.toString().trim() === ""
+        ? t("common.required field text")
+        : ""
+    );
+  };
+
+  useEffect(
+    () => () => {
+      // Detect if the component unmounts
+      isMountedRef.current = false;
+    },
+    []
+  );
 
+  useEffect(() => {
+    checkPostalCodeErrors();
+    checkHouseNumberErrors();
+    // eslint-disable-next-line
+  }, []);
+
+  // Graphql expects the `houseNumberFull` and `postalCode` are defined and for performance `postalCode` should be validated
+  const readyForQuery = !!(houseNumberFull && isValidPostalcode(postalCode));
   const variables = {
     extraHouseNumberFull: "",
     houseNumberFull: sanitizeHouseNumberFull(houseNumberFull),
     postalCode,
     queryExtra: false,
   };
 
-  // Validate forms
-  const validate = (
-    name: string,
-    value?: number | string,
-    required?: boolean
-  ) => {
-    if (touched[name]) {
-      if (required && (!value || value?.toString().trim() === "")) {
-        return t("common.required field text");
-      }
-      if (name === "postalCode" && !isValidPostalcode(value?.toString())) {
-        return t("common.no valid postalcode");
-      }
-    }
-    return undefined;
-  };
-
-  // Error messages
-  const houseNumberError = validate("houseNumber", houseNumber, true);
-  const postalCodeError = validate("postalCode", postalCode, true);
-
-  // Handle all the keys
-  const handleKeyDown = useCallback(
-    (event) => {
-      switch (event.key) {
-        case "Enter":
-          // Prevent submitting the form when pressing Enter
-          event.preventDefault();
-          // Disable focus on the input so the AutoSugest list disappears
-          (document.activeElement as HTMLElement).blur();
-          setFocus(false);
-          break;
-
-        default:
-          // Reset the autoSuggestValue on each key press to prevent bugs
-          setAutoSuggestValue("");
-          break;
-      }
-    },
-    [setFocus]
-  );
-
-  /* There is an issue with `skip`, it's not working if variables are given
-     in `options` to `useQuery`. See https://github.com/apollographql/react-apollo/issues/3367
-     Workaround is not giving any variables if the query should be skipped. */
-  const skip = !!(
-    houseNumberError ||
-    !houseNumberFull ||
-    !postalCode ||
-    !isValidPostalcode(postalCode) ||
-    postalCodeError
-  );
   const { loading, error: graphqlError, data } = useQuery(findAddress, {
-    variables: skip ? undefined : variables,
```

Comment:
Are you sure? Now we can have skip = true but still pass the variables. Is het bug actually fixed?

Reply:
The bug looks fixed when upgrading `@apollo/client` to `3.3.12`.